src/extension/op_def.rs

```diff
@@ -96,10 +95,14 @@ pub(super) enum SignatureFunc {
     // CustomSignatureFunc trait too, and replace this enum with Box<dyn CustomSignatureFunc>.
     // However instead we treat all CustomFunc's as non-serializable.
     #[serde(rename = "signature")]
-    TypeScheme(OpDefTypeScheme),
+    TypeScheme(PolyFuncType),
     #[serde(skip)]
     CustomFunc {
-        params: Vec<TypeParam>,
+        /// All type parameters (both those passed to [func] and any passed
+        /// to the PolyFuncType returned by that)
+        all_params: Vec<TypeParam>,
```

Comment:
Maybe clarify that the first `num_static` params go to func and the remaining ones to the resulting `PolyFuncType`

Reply:
Ok. A bit repetitive given the docstring on `num_static_params` but harmless enough!